### PR TITLE
New changes case collaboration component

### PR DESCRIPTION
**Pull request by vkkurre**
1. Removed all Edit/Delete Option.
2. Added Custom Labels for Placeholder Text, Error Message and Success Message.

### CODE REVIEW

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 1]
```diff
@@ -25,33 +25,29 @@ public class DEL_CaseCollaborationController {
         @AuraEnabled public Boolean blnIsSuccess;
         @AuraEnabled public String strErrorMessage;
         @AuraEnabled public List<DEL_CaseComment__c> list_CaseComments;
-        
+        @AuraEnabled public Map<Id, List<ContentVersion>> list_Attachments;
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:28]:
```suggestion
        @AuraEnabled public Map<Id, List<ContentVersion>> map_AttachmentsByCaseCommentId;
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 2]
```diff
@@ -25,33 +25,29 @@ public class DEL_CaseCollaborationController {
...
+    * @ author       :  Rakesh Nayak & Vinaykant
     * @ description  :  This method queries and returns the CaseComment records related to the case with Id 'recordID'
     * @ params       :  'strRecordId' - Record Id of the Case
     * @ return       :  List of DEL_CaseComment__c records
     **/
     @AuraEnabled(cacheable = true)
     public static ResponseWrapper fetchComments(String strRecordId) {
         ResponseWrapper objResponseWrapper = new ResponseWrapper();
+        Map<Id, ContentDocumentLink> map_ContentDocumentLinkByContentDocumentId = new Map<Id, ContentDocumentLink>();
+        Map<Id, List<ContentVersion>> map_ContentVersionByCaseCommentId = new Map<Id, List<ContentVersion>>();
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:48]:
```suggestion
        Map<Id, List<ContentVersion>> map_ContentVersionsByCaseCommentId = new Map<Id, List<ContentVersion>>();
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 3]
```diff
@@ -25,33 +25,29 @@ public class DEL_CaseCollaborationController {
...
+        List<Id> list_caseCommentIds = new List<Id>();
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:49]:
```suggestion
        List<Id> list_CaseCommentIds = new List<Id>();
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 4]
```diff
@@ -25,33 +25,29 @@ public class DEL_CaseCollaborationController {
...
+        List<Id> list_fileIds = new List<Id>();
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:50]: remove this if not used anywhere

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 5]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
                                                     WHERE Case__c = :strRecordId 
                                                     ORDER BY CreatedDate 
                                                     DESC];
+            
+            if (objResponseWrapper.list_CaseComments.size() > 0) {
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:66]:
```suggestion
            if (!objResponseWrapper.list_CaseComments.ieEmpty()) {
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 6]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
...
+                for (DEL_CaseComment__c objCaseComment : objResponseWrapper.list_CaseComments) {
+                    list_caseCommentIds.add(objCaseComment.Id);
+                }
+                
+                List<ContentDocumentLink> list_contentDocumentLinks = [SELECT
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:71]:
```suggestion
                List<ContentDocumentLink> list_ContentDocumentLinks = [SELECT 
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 7]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
...
+                }
+                
+                List<ContentDocumentLink> list_contentDocumentLinks = [SELECT 
+                                                                       ContentDocumentId,
+                                                                       LinkedEntityId
+                                                                       FROM ContentDocumentLink 
+                                                                       WHERE LinkedEntityId 
+                                                                       IN :list_caseCommentIds];
+                
+                if (list_contentDocumentLinks.size() > 0) {
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:78]:
```suggestion
                if (!list_contentDocumentLinks.isEmpty()) {
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 8]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
...
+                    for (ContentDocumentLink objContentDocumentLink : list_contentDocumentLinks) {
+                        map_ContentDocumentLinkByContentDocumentId.put(objContentDocumentLink.ContentDocumentId, objContentDocumentLink);
+                    }
+                    List<ContentVersion> list_contentVersions = [SELECT
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:82]:
```suggestion
                    List<ContentVersion> list_ContentVersions = [SELECT 
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 9]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
...
+                                                                Id, 
+                                                                FileExtension, 
+                                                                Title, 
+                                                                ContentSize, 
+                                                                CreatedDate, 
+                                                                FileType, 
+                                                                ContentBodyId 
+                                                                FROM ContentVersion 
+                                                                WHERE ContentDocumentId 
+                                                                IN : map_ContentDocumentLinkByContentDocumentId.KeySet()
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:93]:
```suggestion
                                                                IN :map_ContentDocumentLinkByContentDocumentId.KeySet()
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 10]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
...
+                                                                ORDER BY Title];
+                    
+                    for (ContentVersion objContentVersion : list_contentVersions) {
+                        String strCommentId = map_ContentDocumentLinkByContentDocumentId.get(objContentVersion.ContentDocumentId).LinkedEntityId;
+                        if (map_ContentVersionByCaseCommentId.containsKey(strCommentId)) {
+                            map_ContentVersionByCaseCommentId.get(strCommentId).addAll(new List<ContentVersion> { objContentVersion } );
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:99]:
```suggestion
                            map_ContentVersionByCaseCommentId.get(strCommentId).add(objContentVersion);
```

File: Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls [hunk 11]
```diff
@@ -66,6 +62,49 @@ public class DEL_CaseCollaborationController {
...
+                        } else {
+                            map_ContentVersionByCaseCommentId.put(strCommentId, new List<ContentVersion> { objContentVersion } );
+                        }
+                    }
+                }
```
Comment (csankit0108) [inline Delmar/force-app/main/default/classes/DEL_CaseCollaborationController.cls:104]: add new line after this

File: Delmar/force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 12]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomLabels xmlns="http://soap.sforce.com/2006/04/metadata">
```
Comment (csankit0108) [inline Delmar/force-app/main/default/labels/CustomLabels.labels-meta.xml:2]: update the custom label names as mentioned to you.

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.html [hunk 13]
```diff
@@ -72,27 +74,45 @@
...
+                                                            <lightning-layout multiple-rows="true">
+                                                                <template for:each={objCaseComment.listAttachments} for:item="objFile">
+                                                                    <lightning-layout-item size="12" key={objFile.ContentDocumentId} small-device-size="6" medium-device-size="6" 
+                                                                        large-device-size="4" class="slds-p-vertical_xx-small slds-p-right_xx-small">
+                                                                        <lightning-layout class="del-style-file-pills slds-p-around_x-small">
+                                                                            <lightning-layout-item size="1">
+                                                                                <span>
+                                                                                    <lightning-icon icon-name="doctype:attachment" alternative-text="Download" title={objFile.Title}
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.html:86]: you can "Download" also as a custom label

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js [hunk 14]
```diff
@@ -47,46 +35,59 @@ export default class Del_caseCollaborationComponent extends NavigationMixin(Ligh
         this.list_WiredComments = result;
         if (data) {
             if (data.blnIsSuccess) {
+                let objAttachmentList = JSON.parse(JSON.stringify(data.list_Attachments));
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:38]:
```suggestion
                let list_Attachments = JSON.parse(JSON.stringify(data.list_Attachments));
```

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js [hunk 15]
```diff
@@ -47,46 +35,59 @@ export default class Del_caseCollaborationComponent extends NavigationMixin(Ligh
...
+                for (var commentIdKey in objAttachmentList) {
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:39]:
```suggestion
                for (var idCaseCommentId in objAttachmentList) {
```

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js [hunk 16]
```diff
@@ -47,46 +35,59 @@ export default class Del_caseCollaborationComponent extends NavigationMixin(Ligh
...
+                    let list_eachCommentAttachmentList = objAttachmentList[commentIdKey];
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:40]:
```suggestion
                    let list_AttachmentsTemp = objAttachmentList[commentIdKey];
```

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js [hunk 17]
```diff
@@ -47,46 +35,59 @@ export default class Del_caseCollaborationComponent extends NavigationMixin(Ligh
...
+                    for (let objAttachmentOfComment of list_eachCommentAttachmentList) {
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:41]:
```suggestion
                    for (let objAttachment of list_eachCommentAttachmentList) {
```

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js [hunk 18]
```diff
@@ -47,46 +35,59 @@ export default class Del_caseCollaborationComponent extends NavigationMixin(Ligh
...
+                        objAttachmentOfComment["strDownloadURL"] = DEL_FileDownloadURL+objAttachmentOfComment.ContentDocumentId;
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:42]:
```suggestion
                        objAttachmentOfComment["strDownloadURL"] = DEL_FileDownloadURL + objAttachmentOfComment.ContentDocumentId;
```

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js (hunk 15 above)
Comment (csankit0108) [reply Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:39]: have a comment mentioning what we are doing in this loop before staring the loop

File: Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js [hunk 19]
```diff
@@ -47,46 +35,59 @@ export default class Del_caseCollaborationComponent extends NavigationMixin(Ligh
...
                       used to insert the entered comment and refresh the updated comment list
     **/
-
     handleOnClick() {
-        if (this.validateComment()) {
+        this.blnValid = this.validateComment();
+        if (this.blnValid) {
             this.handleIsLoading(true);
-            this.strBody = this.template.querySelector(".nullify").value;
+            //this.strBody = this.template.querySelector(".nullify").value;
```
Comment (csankit0108) [inline Delmar/force-app/main/default/lwc/del_caseCollaborationComponent/del_caseCollaborationComponent.js:90]: remove this line instead of commenting

**Review by csankit0108 — CHANGES_REQUESTED** [review]

File: Delmar/force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 12 above)
Comment (vkkurre) [reply Delmar/force-app/main/default/labels/CustomLabels.labels-meta.xml:2]: Okay Ankit.

**Review by csankit0108 — APPROVED** [review]

**Pull request merged**